### PR TITLE
Make PaymentProcessor use latest SmartContractInterface

### PR DESCRIPTION
**Pull request by Krigpl**
In particular SmartContractInterface is just a simple interface without any hidden logic like GNT to GNTW conversion, so this has been moved to `GNTConverter`.
Also we don't need to pass private key to PaymentProcessor anymore.
And some other minor clean ups, like consistent types for transaction hashes or simplified interfaces.

### CODE REVIEW

File: golem/ethereum/gntconverter.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import logging
+
+from ethereum.utils import denoms
+
+log = logging.getLogger("golem.gnt_converter")
+
+
+class GNTConverter:
+    REQUIRED_CONFS = 2
```
Comment (etam) [inline golem/ethereum/gntconverter.py:9]: Is it possible to have `is_converting` and `is_updating_personal_deposit_address` that do not change the state of object? I understand how this works, but it's quite confusing to see that `is_converting` does some work and `update_personal_deposit_address` being used in an `if` statement.

Comment (Krigpl) [reply golem/ethereum/gntconverter.py:9]: `is_converting` would be split into two methods and the usage would look like:
```
if converter.is_converting():
  converter.keep_converting()
```
I dunno whether it's better. 

`update_personal_deposit_address` suggests modifying the state and imo it's fine to return bool to indicate whether something has been updated or not, what's the issue here?

**Review by etam — APPROVED** [review]

**Comment by etam** [comment]
mypy is sad

**Comment by Krigpl** [comment]
mypy is sad because it thinks it can do static analysis but it can't

**Comment by etam** [comment]
> mypy is sad because it thinks it can do static analysis but it can't

What?

Here, done it for you:

> golem/ethereum/paymentprocessor.py:175: error: Invalid tuple literal type

Use `Tuple[..., ...]` instead of `(..., ...)`.

> golem/ethereum/paymentprocessor.py:183: error: Unsupported left operand type for - ("None")

Add type comment to variable initialization in constructor.

> golem/ethereum/paymentprocessor.py:71: error: Name 'Dict' is not defined

Add `from typing import Dict`.

**Comment by Krigpl** [comment]
I was referring to the only one error caused by this PR which is this one:
> golem/ethereum/paymentprocessor.py:183: error: Unsupported left operand type for - ("None")

While your solution will work I don't think it's valid to just assume this variable will always be None.

**Pull request merged**